processors/token_transfer/contract_events.go

```diff
@@ -105,11 +193,41 @@ func (p *EventsProcessor) parseEvent(tx ingest.LedgerTransaction, opIndex *uint3
 	return protoEvent, nil
 }
 
-// parseCustomTokenEvent attempts to parse a generic SEP41 token event
-func parseCustomTokenEvent(
+// getExpectedTopicsCount returns expected number of topics for each event type based on tx meta version
+func getExpectedTopicsCount(eventType string, txMetaVersion int32) int {
+	switch txMetaVersion {
+	case 3:
+		// V3 format includes admin addresses
+		switch eventType {
+		case TransferEvent:
+			return 4 // ["transfer", from, to, asset]
+		case MintEvent:
+			return 4 // ["mint", admin, to, asset]
+		case ClawbackEvent:
+			return 4 // ["clawback", admin, from, asset]
+		case BurnEvent:
+			return 3 // ["burn", from, asset]
+		}
```

Comment:
I understand the verbosity but would you be opposed to
```suggestion
		// V3 format includes admin addresses
		switch eventType {
		case BurnEvent:
			return 3 // ["burn", from, asset]
		default:
			// ["transfer", from, to, asset]
			// ["mint", admin, to, asset]
			// ["clawback", admin, from, asset]
			return 4
		}
```
?